### PR TITLE
tickr: init at 0.7.1

### LINKED ISSUE #308304: Package request: Tickr
**Project description**

GTK-based highly graphically-customizable Feed Ticker

**Metadata**

* homepage URL: http://www.open-tickr.net/
* source URL: http://www.open-tickr.net/src/tickr-0.7.1.tar.gz
* license: GPLv3
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by StarryReverie**
## Description of changes

Closes #308304.

`tickr` is a GTK-based highly graphically-customizable feed ticker. Homepage: <http://www.open-tickr.net>.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ti/tickr/package.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+, libxml2
+, pkg-config
+}:
+
+stdenv.mkDerivation {
+  pname = "tickr";
+  version = "0.7.1";
+
+  src = fetchurl {
+    url = "http://www.open-tickr.net/src/tickr-0.7.1.tar.gz";
```
Comment (fabaff) [inline pkgs/by-name/ti/tickr/package.nix:17]:
```suggestion
    url = "http://www.open-tickr.net/src/tickr-${version}.tar.gz";
```

File: pkgs/by-name/ti/tickr/package.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  ];
+
+  patchPhase = ''
+    runHook prePatch
+
+    substituteInPlace src/tickr/tickr.h \
+        --replace-fail "/usr/share" "$out/share"
+
+    runHook postPatch
+  '';
```
Comment (fabaff) [inline pkgs/by-name/ti/tickr/package.nix:39]:
```suggestion
  postPatch = ''
    substituteInPlace src/tickr/tickr.h \
        --replace-fail "/usr/share" "$out/share"
  '';
```

File: pkgs/by-name/ti/tickr/package.nix [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  buildPhase = ''
+    runHook preBuild
+
+    # -Wformat is required by -Wsecurity-format
+    NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -Wformat" make
+
+    runHook post
+  '';
```
Comment (fabaff) [inline pkgs/by-name/ti/tickr/package.nix:48]: Perhaps `makeFlags` can help here.

File: pkgs/by-name/ti/tickr/package.nix [hunk 4]
```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchurl
+
+, fribidi
+, gnutls
+, gtk2
```
Comment (Aleksanaa) [inline pkgs/by-name/ti/tickr/package.nix:7]: This PR may not be merged because we are already actively removing gtk2 packages.

**Pull request closed without merging**